hypothesis-python/src/hypothesis/internal/conjecture/shrinking/floats.py

```diff
@@ -14,14 +14,28 @@
 from hypothesis.internal.conjecture.floats import float_to_lex
 from hypothesis.internal.conjecture.shrinking.common import Shrinker
 from hypothesis.internal.conjecture.shrinking.integer import Integer
+from hypothesis.internal.floats import sign_aware_lte
 
 MAX_PRECISE_INTEGER = 2**53
 
 
 class Float(Shrinker):
-    def setup(self):
+    def setup(self, node):
         self.NAN = math.nan
         self.debugging_enabled = True
+        self.node = node
+
+    def consider(self, value):
+        min_value = self.node.kwargs["min_value"]
+        max_value = self.node.kwargs["max_value"]
+        if not math.isnan(value) and not (
+            sign_aware_lte(min_value, value) and sign_aware_lte(value, max_value)
+        ):
```

Comment:
Check against `smallest_nonzero_magnitude` too, for subnormals?  (yes, they're really annoying!)

Reply:
this comment sent me down a rabbit hole of implementing `ir_value_permitted` for all ir types, not just float. I think the up-front effort here will pay off when working on future shrink passes. We get to use `ir_value_permitted` both here and when re-aligning trees.

One of these days I'll consolidate all these ir-related functions into a proper `IRType` class with our five concrete instantiations...